Core/src/exception.hpp

```diff
@@ -0,0 +1,91 @@
+/*! Definition of the base exception class for the analysis namespace.
+This file is part of https://github.com/hh-italian-group/TauMLTools. */
+
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <sstream>
+#include <boost/format.hpp>
+#include <boost/stacktrace.hpp>
```

Comment:
Last time I checked, with stacktrace.hpp the code was not compiling with CMSSW. Please, check that `scram build` works.